app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -100,20 +105,24 @@ class BrowserWebViewClient(
                 }
                 is SpecialUrlDetector.UrlType.Unknown -> {
                     Timber.w("Unable to process link type for ${urlType.url}")
-                    webView.loadUrl(webView.originalUrl)
+                    webView.loadUrl(webView.originalUrl, globalPrivacyControl.getHeaders())
                     false
                 }
                 is SpecialUrlDetector.UrlType.SearchQuery -> false
                 is SpecialUrlDetector.UrlType.Web -> {
                     if (requestRewriter.shouldRewriteRequest(url)) {
                         val newUri = requestRewriter.rewriteRequestWithCustomQueryParams(url)
-                        webView.loadUrl(newUri.toString())
+                        webView.loadUrl(newUri.toString(), globalPrivacyControl.getHeaders())
                         return true
                     }
                     if (isForMainFrame) {
                         webViewClientListener?.willOverrideUrl(url.toString())
+                        if (globalPrivacyControl.isGpcActive()) {
+                            webView.loadUrl(url.toString(), globalPrivacyControl.getHeaders())
```

Comment:
I see two potential gotchas here:
* this could get called for subframes on lower APIs and potentially allow an ad banner to be loaded fullscreen or break a payment page. Perhaps we should only support this for API 24 and above where we can be sure the request is from the main frame.
* we could end up in an endless loop. Perhaps we could check that the header isn't already there. We get the full request from API 24

Reply:
Hey @subsymbolic I've been exploring what you said about "subframes", but I've not been able to find any scenario where that method is called from an iframe. 🤷‍♀️  I'm still trying to find something but without any luck. For `shouldInterceptRequest` we receive requests that are coming from a subframes, but not in the other method.

For example, I've playing around using this: https://jsfiddle.net/cmonfortep/d1poxc75/12/ where I refresh the iframe. I see the request via `shouldInterceptRequest`, but not on the `shouldOverrideUrlLoading`.

Reply:
I've been playing with user agents, requests, etc this week and gut feeling says to do all this in `shouldInterceptRequest` same way that we do the https update instead of `shouldOverrideUrlLoading` as I initially thought.

Reply:
I've tried to move the code around like I said to `shouldInterceptRequest` and made some changes on a separate branch `feature/marcos/gpc_default_on` based on what I've been doing to change the user agent on a separate project. I think it works or at least is pretty close, it definitely doesn't crash on a _blank link.

@cmonfortep we'll catch up on Monday because I'm not 100% convinced if my version is breaking https://duckduckgo.github.io/privacy-test-pages/security/spoof-js-page-rewrite.html (I kinda think is not)

Reply:
Closing this conversation, this code is now in `shouldInterceptRequest` plus it checks if the header already exists to avoid an infinite loop like @subsymbolic said which I was able to reproduce.

Reply:
Just saw the earlier message to me, so sorry I missed that. Glad this is resolved now 👍